From b84054fb216ccf322a970cabdde2b0bd34f60d5c Mon Sep 17 00:00:00 2001
From: "kalyan.kondapally@intel.com"
 <kalyan.kondapally@intel.com@0039d316-1c4b-4281-b951-d872f2087c98>
Date: Thu, 27 Feb 2014 13:21:11 +0000
Subject: [PATCH] Fix potential crash in CursorLoaderOzone.

We load a different set of cursors with Ash and DesktopAura. Hence,
we cannot assume kCursorPointer is always loaded. Instead of making
any assumptions for default cursor image, we set this as NULL.

Review URL: https://codereview.chromium.org/179903006

git-svn-id: svn://svn.chromium.org/chrome/trunk/src@253775 0039d316-1c4b-4281-b951-d872f2087c98
---
 ui/base/cursor/cursor_loader_ozone.cc | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/ui/base/cursor/cursor_loader_ozone.cc b/ui/base/cursor/cursor_loader_ozone.cc
index bbe524f..c853a3b 100644
--- a/ui/base/cursor/cursor_loader_ozone.cc
+++ b/ui/base/cursor/cursor_loader_ozone.cc
@@ -64,11 +64,9 @@ void CursorLoaderOzone::SetPlatformCursor(gfx::NativeCursor* cursor) {
     // TODO(dnicoara) Add support for custom cursors: crbug.com/343155
     cursor->SetPlatformCursor(cursor->platform());
   } else {
-    const gfx::ImageSkiaRep& image_rep =
-        cursors_[kCursorPointer]->GetRepresentation(
-            display().device_scale_factor());
-
-    cursor->SetPlatformCursor(&image_rep.sk_bitmap());
+    // We load different set of cursors with Ash and DesktopAura. Since we do
+    // not have the resource for this cursor, set platform cursor to NULL.
+    cursor->SetPlatformCursor(NULL);
   }
 }

--
1.9.0
